src/Dots/Dots.jsx

```diff
@@ -3,29 +3,36 @@ import React, { Component, PropTypes } from 'react'
 import Dot from './Dot'
 import dynStyle from './style'
 
-const maxDots = 3
-const maxOpacity = 0.6
-const minOpacity = 0.1
-const timeout = 1000
+const MAX_DOTS = 3
+const MAX_OPACITY = 0.6
+const MIN_OPACITY = 0.1
+const TIMEOUT = 1000
 
 class Dots extends Component {
   constructor(props, context) {
     super(props, context)
 
     this.state = {
-      opacities: Array.from(Array(maxDots)).map(() => minOpacity),
+      opacities: Array.from(Array(MAX_DOTS)).map(() => MIN_OPACITY),
     }
 
     this.intervals = []
+    this.timeouts = []
   }
 
   componentDidMount() {
-    for (let i = 0; i < maxDots; ++i) {
-      setTimeout(() => this.runDot(i), (timeout / maxDots) * i)
+    for (let i = 0; i < MAX_DOTS; ++i) {
+      this.timeouts.push(setTimeout(() => this.runDot(i), (TIMEOUT / MAX_DOTS) * i))
     }
   }
 
   componentWillUnmount() {
+    if (this.timeouts) {
```

Comment:
Is this condition necessary ? If timeouts is empty, i guess the "for" statement won't be processed.